src/azure-cli/azure/cli/command_modules/synapse/operations/sqlpoolblobauditingpolicy.py

```diff
@@ -0,0 +1,229 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=unused-argument
+from azure.mgmt.synapse.models import BlobAuditingPolicyState
+from azure.cli.core.commands.client_factory import get_mgmt_service_client
+from azure.cli.core.profiles import ResourceType
+from knack.util import CLIError
+
+
+def sqlpool_blob_auditing_policy_update(
+        cmd,
+        instance,
+        state=None,
+        storage_account=None,
+        storage_endpoint=None,
+        storage_account_access_key=None,
+        retention_days=None,
+        audit_actions_and_groups=None,
+        storage_account_subscription_id=None,
+        is_storage_secondary_key_in_use=None,
+        is_azure_monitor_target_enabled=None):
+    '''
+    Updates a sql pool blob auditing policy. Custom update function to apply parameters to instance.
+    '''
+
+    # Validate input arguments
+    blob_storage_arguments_provided = (
+        storage_account is not None or
+        storage_endpoint is not None or
+        storage_account_access_key is not None or
+        retention_days is not None)
+
+    if not state and not blob_storage_arguments_provided:
+        raise CLIError('Either state or blob storage arguments are missing')
```

Comment:
Do you mean user must provide at least one parameter from below list? Can we use `any()`  like:

```
if not any([storage_account, storage_endpoint, storage_account_access_key, retention_days, state]):
    raise CLIError('Either state or blob storage arguments are missing')
```

Reply:
Thanks for reminder, I refactor the `blob_storage_arguments_provided` by using `any`.

---
In reply to: [533138845](https://github.com/Azure/azure-cli/pull/16095#discussion_r533138845) [](ancestors = 533138845)